lib/elixir/lib/regex.ex

```diff
@@ -412,6 +412,80 @@ defmodule Regex do
     opts
   end
 
+  @doc """
+  Returns the pattern as an embeddable string.
+
+  If the pattern was compiled with an option which cannot be represented
+  as an embeddable modifier in the current version of PCRE and strict is true
+  (the default) then an ArgumentError exception will be raised.
+
+  When strict is false the pattern will be returned as though any offending
+  options had not be used and the function will not raise any exceptions.
+
+  Embeddable modifiers/options are currently:
+    * 'i' - :caseless
+    * 'm' - :multiline
+    * 's' - :dotall, :newline, :anycrlf})
+    * 'x' - :extended
+
+  And unembeddable modifiers are
+    * 'f' - :firstline
+    * 'U' - :ungreedy
+    * 'u' - :unicode, :ucp
+
+  Any other regex compilation option not listed here is considered unembeddable.
+
+  ## Examples
+      iex> Regex.to_embed(~r/foo/)
+      "(?-imsx:foo)"
+
+      iex> Regex.to_embed(~r/^foo/m)
+      "(?m-isx:^foo)"
+
+      iex> Regex.to_embed(~r/foo # comment/ix)
+      "(?ix-ms:foo # comment\\n)"
+
+      iex> Regex.to_embed(~r/foo/iu)
+      ** (ArgumentError) regex compiled with options [:ucp, :unicode] which cannot be represented as an embedded pattern in this version of PCRE
+
+      iex> Regex.to_embed(~r/foo/imsxu, strict: false)
+      "(?imsx:foo\\n)"
+
+  """
+  @spec to_embed(t, [term]) :: String.t()
+  def to_embed(%Regex{source: source, opts: regex_opts}, embed_opts \\ []) do
+    strict = Keyword.get(embed_opts, :strict, true)
+
+    modifiers =
+      case embeddable_modifiers(regex_opts) do
+        {:ok, modifiers} ->
+          modifiers
+
+        {:error, modifiers, untranslatable} ->
+          if strict do
+            raise ArgumentError,
+                  "regex compiled with options #{inspect(untranslatable)} which cannot be " <>
+                    "represented as an embedded pattern in this version of PCRE"
+          else
+            modifiers
+          end
+      end
+
+    disabled =
+      Enum.reject([?i, ?m, ?s, ?x], &(&1 in modifiers))
+      |> List.to_string()
```

Comment:
```suggestion
    disabled = List.to_string([?i, ?m, ?s, ?x] -- modifiers)
```

Btw, you don't need the `List.to_string` here and below. You can interpolate a list of characters into a regex. :)

Reply:
Oh, neat, i didnt know this syntax. Thanks. Ill remove the to_string as well then in a follow up.